src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -284,6 +286,119 @@ public bool PrefixReserved
+        {
+            get { return _packageMetadata?.DeprecationMetadata != null; }
+        }
+
+        public bool PackageDeprecationAlternatePackageRangeHasUpperBound
+        {
+            get { return _packageMetadata?.DeprecationMetadata?.AlternatePackage?.Range?.HasUpperBound ?? false; }
+        }
+
+        private Uri _packageDeprecationAlternatePackageLinkUrl;
+        public Uri PackageDeprecationAlternatePackageLinkUrl
+        {
+            get { return _packageDeprecationAlternatePackageLinkUrl; }
+            set
+            {
+                if (_packageDeprecationAlternatePackageLinkUrl != value)
+                {
+                    _packageDeprecationAlternatePackageLinkUrl = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationAlternatePackageLinkUrl));
+                }
+            }
+        }
+
+        private string _packageDeprecationReasons;
+        public string PackageDeprecationReasons
+        {
+            get { return _packageDeprecationReasons; }
+            set
+            {
+                if (_packageDeprecationReasons != value)
+                {
+                    _packageDeprecationReasons = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationReasons));
+                }
+            }
+        }
+
+        private string _packageDeprecationReasonsExplained;
+        public string PackageDeprecationReasonsExplained
+        {
+            get { return _packageDeprecationReasonsExplained; }
+            set
+            {
+                if (_packageDeprecationReasonsExplained != value)
+                {
+                    _packageDeprecationReasonsExplained = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationReasonsExplained));
+                }
+            }
+        }
+
+        private string _packageDeprecationCustomMessage;
+        public string PackageDeprecationCustomMessage
+        {
+            get { return _packageDeprecationCustomMessage; }
+            set
+            {
+                if (_packageDeprecationCustomMessage != value)
+                {
+                    _packageDeprecationCustomMessage = value;
+                    OnPropertyChanged(nameof(PackageDeprecationCustomMessage));
+                }
+            }
+        }
+
+        private string _packageDeprecationAlternatePackageLinkText;
+        public string PackageDeprecationAlternatePackageLinkText
+        {
+            get { return _packageDeprecationAlternatePackageLinkText; }
+            set
+            {
+                if (_packageDeprecationAlternatePackageLinkText != value)
+                {
+                    _packageDeprecationAlternatePackageLinkText = value;
+
+                    OnPropertyChanged(nameof(PackageDeprecationAlternatePackageLinkText));
+                }
+            }
+        }
+
+        public string ExplainPackageDeprecationReasons(IEnumerable<string> reasons)
```

Comment:
nit: this should probably be stored as a set or enum as soon as possible. multiple `Contains` on `IEnumerable` could be re-materializing things and is a linear search.

Reply:
Done in https://github.com/NuGet/NuGet.Client/pull/2899/commits/f5c101bc81de17aa1060d51aa49b19bec14097cc